packages/ckeditor5-utils/src/focustracker.ts

```diff
@@ -85,23 +162,77 @@ export default class FocusTracker extends /* #__PURE__ */ DomEmitterMixin( /* #_
 			throw new CKEditorError( 'focustracker-add-element-already-exist', this );
 		}
 
-		this.listenTo( element, 'focus', () => this._focus( element ), { useCapture: true } );
-		this.listenTo( element, 'blur', () => this._blur(), { useCapture: true } );
+		this.listenTo( element, 'focus', () => {
+			// @if CK_DEBUG_FOCUSTRACKER // console.log( `"${ getName( this ) }": Focus with useCapture on DOM element` );
+
+			const externalFocusedViewInSubtree = this.externalViews.find( view => isExternalViewSubtreeFocused( element, view ) );
+
+			if ( externalFocusedViewInSubtree ) {
+				this._focus( externalFocusedViewInSubtree.element! );
+			} else {
+				this._focus( element );
+			}
+		}, { useCapture: true } );
+
+		this.listenTo( element, 'blur', () => {
+			// @if CK_DEBUG_FOCUSTRACKER // console.log( `"${ getName( this ) }": Blur with useCapture on DOM element` );
+
+			this._blur();
+		}, { useCapture: true } );
+
 		this._elements.add( element );
 	}
 
 	/**
-	 * Stops tracking the specified element and stops listening on this element.
+	 * Removes a DOM element from the focus tracker.
 	 */
-	public remove( element: Element ): void {
+	private _removeElement( element: Element ): void {
+		if ( this._elements.has( element ) ) {
+			this.stopListening( element );
+			this._elements.delete( element );
+		}
+
 		if ( element === this.focusedElement ) {
 			this._blur();
 		}
+	}
 
-		if ( this._elements.has( element ) ) {
-			this.stopListening( element );
-			this._elements.delete( element );
+	/**
+	 * Adds an external `FocusTracker` instance to this focus tracker and makes it contribute to this focus tracker's state.
+	 */
+	private _addView( view: ViewWithFocusTracker ): void {
+		if ( view.element ) {
```

Comment:
Well, okay, maybe I went too much into thinking about linking focus trackers.

But if you think from the perspective that "I am just adding a view to a focus tracker (and don't know if it has FT itself / I don't care / I don't know about this internal mechanism)" then adding the element makes sense.

Reply:
* One of the factors here was backward compatibility. 
   * If we stopped adding `view.element`, then the code that migrated from `this.focusTracker.add( someView.element )` to `this.focusTracker.add( someView )` (for instance `ToolbarView`) would experience some breaking changes. If integrators iterated over `ToolbarView.focusTracker.elements` they wouldn't get the same set of elements as in the past. 
* The other factor was that DOM listeners must stay for functional reasons when adding views with focus trackers.
  * Just because a view has a focus tracker, it does not guarantee it tracks its own `#element` 😀. Yep, this is our architecture, we made no restrictions or requirements designing our framework and we cannot just change it. Knowing that we cannot depend on the fact that the external view's focus tracker will contribute to the state correctly. Also, to make it even more complex, the external view's focus tracker can be dynamic and it can add or remove tracked elements independently so any assumption made here in the code could be invalid if we didn't listen to the DOM element in the olf-fashioned way.
  * One of the tests that fails after removing this one bit is the following one. It fails on L523 because `childViewB`, despite owning a focus tracker instance, only tracks its children. https://github.com/ckeditor/ckeditor5/blob/c44ec5036197e5f7a2b35a0152f62b0eff0e43c0/packages/ckeditor5-utils/tests/focustracker.js#L500-L530